### PR TITLE
fixes #350, #366, #373

### LINKED ISSUE #350: "AttributeError: 'list' object has no attribute 'values'" on Jira Error handling
Running jira library 1.0.10 on JIRA (v7.3.1#73012-sha1:68837e3) 
```
Traceback (most recent call last):
  File "/Users/emmanuelsciara/virtual_envs/propagate_issues/lib/python3.5/site-packages/behave/model.py", line 1456, in run
    match.run(runner.context)
  File "/Users/emmanuelsciara/virtual_envs/propagate_issues/lib/python3.5/site-packages/behave/model.py", line 1903, in run
    self.func(context, *args, **kwargs)
  File "features/steps/basic.py", line 39, in when_propagate_issues_to_target
    context.jira.create_issues(issue_list)
  File "/Users/emmanuelsciara/virtual_envs/propagate_issues/lib/python3.5/site-packages/jira/client.py", line 936, in create_issues
    r = self._session.post(url, data=json.dumps(data))
  File "/Users/emmanuelsciara/virtual_envs/propagate_issues/lib/python3.5/site-packages/jira/resilientsession.py", line 153, in post
    return self.__verb('POST', url, **kwargs)
  File "/Users/emmanuelsciara/virtual_envs/propagate_issues/lib/python3.5/site-packages/jira/resilientsession.py", line 146, in __verb
    raise_on_error(response, verb=verb, **kwargs)
  File "/Users/emmanuelsciara/virtual_envs/propagate_issues/lib/python3.5/site-packages/jira/resilientsession.py", line 49, in raise_on_error
    error_list = response['errors'].values()
AttributeError: 'list' object has no attribute 'values'
```
This happen on response:
```
{
    'errors': [
        {
            'status': 400, 'failedElementNumber': 0,
            'elementErrors': {
                'errorMessages': [],
                'errors': {
                    'customfield_10002': "Field 'customfield_10002' cannot be set. It is not on the appropriate screen, or unknown."
                }
            }
        }
    ],
    'issues': []
}
```
Offending code is https://github.com/pycontribs/jira/blob/ff7fe0b8cd901e6d84abafbc10a5eab405568649/jira/resilientsession.py#L47-L50 :
```
                elif 'errors' in response and len(response['errors']) > 0:
                    # JIRA 6.x error messages are found in this array.
                    error_list = response['errors'].values()
                    error = ", ".join(error_list)
```

### PR DESCRIPTION
**Pull request by esciara**
fix #350

### CODE REVIEW

**Comment by esciara** [comment]
@ssbarnea  Travis test fails not due to this code. See #366

File: tests/tests.py [hunk 1]
```diff
@@ -184,6 +184,7 @@ def __init__(self):
                 else:
                     self.CI_JIRA_USER_PASSWORD = 'sd4s3dgec5fhg4tfsds3434'
 
+                self.CI_JIRA_ADMIN_EMAIL_DOMAINE_NAME = os.environ.get('CI_JIRA_ADMIN_EMAIL_DOMAINE_NAME', 'ssbarnea.33mail.com')
```
Comment (ssbarnea) [inline tests/tests.py]: "DOMAIN" not "DOMAINE" ;)

Comment (esciara) [inline tests/tests.py]: :D The french in me... I changed it. Will push when found a solution to what I last reported on #366

**Comment by esciara** [comment]
Hi @ssbarnea: [my build](https://travis-ci.org/esciara/jira/builds/215029379) worked on the last commit and only one job failed on the [pull request build](https://travis-ci.org/pycontribs/jira/builds/215029390), and just needs to be restarted... Ok to review for merge?

Em

**Comment by esciara** [comment]
This pull request should also fix #373 .

File: tests/tests.py (hunk 1 above)
Comment (esciara) [inline tests/tests.py]: corrected.

**Comment by mulmschneider** [comment]
I just got bitten by #350 - can anything be done to get this merged? Would it make sense to split 199f91a1764f37c96c44e239aa2ca9481802d4b8 from the rest of this PR?

**Comment by ssbarnea** [comment]
@mulmschneider @esciara Yes, the smaller a PR is the bigger are the chances of getting it merged. Huge ones are likely to stay forever open.

I will close this one because you are unlikely to rebase it successfully. 

Cherry pick what you need and open a new one. Once it passed CI, mention me on a comment and I will review and merge. I will try to make a new release very other week.

**Pull request closed without merging**